datadog/resource_datadog_cloud_workload_security_agent_rule.go

```diff
@@ -0,0 +1,168 @@
+package datadog
+
+import (
+	"context"
+
+	_ "gopkg.in/warnings.v0"
+
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+
+	datadogV2 "github.com/DataDog/datadog-api-client-go/api/v2/datadog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const agentRuleType = "agent_rule"
+
+func resourceDatadogCloudWorkloadSecurityAgentRule() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a Datadog Cloud Workload Security Agent Rule API resource for agent rules.",
+		CreateContext: resourceDatadogCloudWorkloadSecurityAgentRuleCreate,
+		ReadContext:   resourceDatadogCloudWorkloadSecurityAgentRuleRead,
+		UpdateContext: resourceDatadogCloudWorkloadSecurityAgentRuleUpdate,
+		DeleteContext: resourceDatadogCloudWorkloadSecurityAgentRuleDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: cloudWorkloadSecurityAgentRuleSchema(),
+	}
+}
+
+func cloudWorkloadSecurityAgentRuleSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+
+		"description": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "The description of the Agent rule.",
+		},
+
+		"enabled": {
+			Type:        schema.TypeBool,
+			Required:    true,
+			Description: "Whether the Agent rule is enabled.",
+		},
+
+		"expression": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "The SECL expression of the Agent rule.",
+		},
+
+		"name": {
```

Comment:
It seems you cannot update the `name` attribute. Can you set `ForceNew: true` in this case so that the resource is recreated if this is changed. Otherwise, i believe there would be a perpetual diff if a user changes the name.

Reply:
Done